src/ansys/fluent/core/solver/flobject.py

```diff
@@ -584,8 +589,13 @@ def get_object_names(self):
         return obj_names_list
 
     def __getitem__(self, name: str) -> ChildTypeT:
-        if name not in self.get_object_names():
-            raise KeyError(name)
+        try:
+            if name not in self.get_object_names():
+                raise KeyError(name)
+        except KeyError:
+            keys = self.get_object_names()
+            print(f"KeyError: '{name}'")
+            print(f"Available keys are - {keys}")
```

Comment:
@hpohekar, I think here we can use the "allowed_name_error_msg.py" to display the names that are closely matching the name passed by user.

Reply:
Yes we have integrated both approaches. Thank you.